### PR TITLE
ZENKO-2870 feat: test utils methods and hooks

### PR DESCRIPTION
**Pull request by romainlancia**

### CODE REVIEW

File: src/react/utils/__tests__/hooks.test.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { mount } from 'enzyme';
+
+const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight');
+const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth');
+
+const Component = () => {
+    const ref = useRef(null);
+    const height = useHeight(ref);
+
+    useOutsideClick(ref, () => {
```
Comment (nicolas2bert) [inline src/react/utils/__tests__/hooks.test.jsx:14]: Should not we check that the `useOutsideClick` callback is being called when we click outside of the component?
I did not understand how you test it.

File: src/react/utils/__tests__/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    });
+
+    it('should return a string with a trailing slash added at the end if no exists', () => {
+        urls.forEach(url => {
+            expect(addTrailingSlash(url).slice(-1)).toBe('/');
+        });
+        expect(addTrailingSlash('').slice(-1)).toBe('');
+    });
+
+    it('should return a string which pluralize the noun taken in params in count > 1', () => {
```
Comment (nicolas2bert) [inline src/react/utils/__tests__/index.test.jsx:132]: Can we test the third parameter as well?

File: src/react/utils/__tests__/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        expect(formatBytes(10240000000000000000000000, 1)).toBe('8.5 YB');
+    });
+
+    const urls = ['/buckets/', '/buckets', '/buckets/bucket1', '/buckets/bucket1/test/'];
+
+    it('should return a string with a trailing slash removed at the end if exists', () => {
+        urls.forEach(url => {
+            expect(stripTrailingSlash(url).slice(-1)).not.toBe('/');
+        });
+        expect(stripTrailingSlash('').slice(-1)).not.toBe('/');
```
Comment (nicolas2bert) [inline src/react/utils/__tests__/index.test.jsx]: "Empty" url could have been added to the `urls` array.

File: src/react/utils/__tests__/index.test.jsx (hunk 2 above)
Comment (romainlancia) [reply src/react/utils/__tests__/index.test.jsx:132]: done

File: src/react/utils/__tests__/index.test.jsx (hunk 3 above)
Comment (romainlancia) [reply src/react/utils/__tests__/index.test.jsx]: done

File: src/react/utils/__tests__/hooks.test.jsx (hunk 1 above)
Comment (romainlancia) [reply src/react/utils/__tests__/hooks.test.jsx:14]: done: I was listening events directly from the document but I change that I now watch if useOutsideClick is triggered it much clearer

File: src/react/utils/__tests__/hooks.test.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        expect(component.find('div').text()).toBe('300');
+    });
+
+    it('should trigger if a click occurs outside the component', async () => {
+        const useOutsideClickMock = jest.spyOn(hooks, 'useOutsideClick').mockImplementation(() => {});
+
+        const component = mount(
+            <div>
+                <Component/>
+                <button onClick={() => {
```
Comment (nicolas2bert) [inline src/react/utils/__tests__/hooks.test.jsx]:
```suggestion
                <button onClick={() => {}}>
```

**Review by nicolas2bert — APPROVED** [review]

File: src/react/utils/__tests__/hooks.test.jsx (hunk 4 above)
Comment (romainlancia) [reply src/react/utils/__tests__/hooks.test.jsx]: done

**Review by tmacro — APPROVED** [review]

**Comment by romainlancia** [comment]
@bert-e approve

**Pull request merged**